daemon/create_windows.go

```diff
@@ -1,11 +1,82 @@
 package daemon
 
 import (
+	"fmt"
+
 	"github.com/docker/docker/image"
+	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/runconfig"
+	"github.com/docker/docker/volume"
 )
 
 // createContainerPlatformSpecificSettings performs platform specific container create functionality
 func createContainerPlatformSpecificSettings(container *Container, config *runconfig.Config, hostConfig *runconfig.HostConfig, img *image.Image) error {
+	for spec := range config.Volumes {
+		mp, err := volume.ParseMountSpec(spec, hostConfig.VolumeDriver)
```

Comment:
Gosh, I stared at this for a few minutes, and trying to combine them would IMO create more confusion than anything else. Sure, I don't disagree there's some code in common though.